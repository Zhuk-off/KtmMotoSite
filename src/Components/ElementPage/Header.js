import {
  StyledHeader,
  StyledHeaderText,
} from '../styles/ElementPages.styled/Header.styled';

const Header = (props) => {
  let models;
  switch (props.length) {
    case 1:
      models = 'модель';
      break;
    case 2:
    case 3:
    case 4:
      models = 'модели';
      break;
    default:
      models = 'моделей';
      break;
  }
  if (props.length === undefined) {
    models = '';
  }
  return (
    <StyledHeader bg={'url(' + props.background + ')'}>
      <StyledHeaderText>
        <h1>{props.category} </h1>
        <h2>{props.description}</h2>
        <h3>
          {props.length} {models}
        </h3>
      </StyledHeaderText>
    </StyledHeader>
  );
};

export { Header };
