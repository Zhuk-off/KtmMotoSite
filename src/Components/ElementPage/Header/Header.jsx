import styles from './Header.module.scss';

const Header = (props) => {
  let models;
  switch (props.length) {
    case 1:
      models = 'модель';
      break;
    case 2:
    case 3:
    case 4:
      models = 'модели';
      break;
    default:
      models = 'моделей';
      break;
  }
  if (props.length === undefined) {
    models = '';
  }
  return (
    <div
      className={styles.pageHeading}
      style={{
        backgroundImage: 'url(' + props.background + ')',
      }}
    >
      <div className={styles.pageHeadingText}>
        <h1>{props.category} </h1>
        <h2>{props.description}</h2>
        <h3>
          {props.length} {models}
        </h3>
      </div>
    </div>
  );
};

export { Header };
