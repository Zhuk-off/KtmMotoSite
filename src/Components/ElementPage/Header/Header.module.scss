.pageHeading {
  color: #fff;
  height: 40vh;
  width: 100vw;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.pageHeadingText {
  text-align: center;
  padding-top: 150px;
  letter-spacing: 0.5px;
}
