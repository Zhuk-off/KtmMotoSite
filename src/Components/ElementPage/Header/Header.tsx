import React from 'react'
import styles from './Header.module.scss'
import { FC } from 'react'

interface IHeaderProps {
  category?: string
  description?: string
  length?: number
  background: string
}

const Header: FC<IHeaderProps> = (props) => {
  let models
  switch (props.length) {
    case 1:
      models = 'модель'
      break
    case 2:
    case 3:
    case 4:
      models = 'модели'
      break
    default:
      models = 'моделей'
      break
  }
  if (props.length === undefined) {
    models = ''
  }
  return (
    <div
      className={styles.pageHeading}
      style={{
        backgroundImage: 'url(' + props.background + ')',
      }}
    >
      <div className={styles.pageHeadingText}>
        <h1>{props.category} </h1>
        <h2>{props.description}</h2>
        <h3>
          {props.length} {models}
        </h3>
      </div>
    </div>
  )
}

export { Header }
