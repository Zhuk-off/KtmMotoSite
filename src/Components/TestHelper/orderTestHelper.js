import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

export const emailTestValidate = async (input, typeText, errorText) => {
  input.value = '';
  await userEvent.type(input, typeText);
  expect(screen.getByText(errorText)).toBeInTheDocument();
};
