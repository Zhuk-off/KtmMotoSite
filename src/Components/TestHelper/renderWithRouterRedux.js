import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import store from '../../store/store';

export const renderWithRouterReduxOrder = (component, initialEntries = '/') => {
  return render(
    <Provider store={store}>
      <MemoryRouter initialEntries={[initialEntries]}>
        <Routes>
          <Route path={`naked/:id/order`} element={component} />
        </Routes>
      </MemoryRouter>
    </Provider>
  );
};
