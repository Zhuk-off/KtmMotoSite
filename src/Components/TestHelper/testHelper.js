export const moto = {
  background:
    'http://ktm.by/images/made/images/bg/2713/361975_MY21-KTM-890-DUKE_Action_cebf97f0912754aa122e60151c9c0586.jpeg',
  buy: false,
  category: 'naked',
  counter: 1,
  descriptionMoto:
    'KTM 8 90 DUKE дает новое определение слову «резкий», добавляя дополнительные грани. KTM 890 DUKE, оснащенный рядным двухцилиндровым двигателем объемом 889 куб.см, размещенным на одном из самых легких и компактных шасси, может похвастаться маневренностью, которую вы ожидаете от 600-кубового двигателя, но с мощностью более крупного двигателя. Он не переписывает книгу правил для среднекубатурного "нэйкеда" , он стирает его. Цена - НЕ ЯВЛЯЕТСЯ ФИКСИРОВАННОЙ, согласуется в договорном порядке. Конфигуратор дополнительного оборудования: https://configurator.ktm.com/m...',
  id: 'ktm-890-duke-2022',
  name: 'KTM 890 Duke 2022',
  photolink:
    'http://ktm.by/images/made/images/moto/ktm-790-duke-2018-1/orange/2715/372812_890-Duke-ORANGE-MY21-90-Right_30313d26236d139f4f81ffc0ef58aaf2.jpg',
  price: '42336 руб',
  titleCard: 'Скальпель дорог',
};

export const page = {
  background:
    'http://ktm.by/images/made/images/bg/2473/1290-SUPER-DUKE-R-2020-cover_2f552fd4e6f76672742481ebb95c4027.jpeg',
  categoryTitle: 'KTM Naked Bikes',
  description: 'Городские мотоциклы-стритфайтеры',
  id: 'naked',
  path: '/naked',
};

export const motoState = [
  {
    background:
      'http://ktm.by/images/made/images/bg/2816/358676_KTM-Naked-Model-Range_-Action_cebf97f0912754aa122e60151c9c0586.jpeg',
    buy: false,
    category: 'naked',
    counter: 1,
    descriptionMoto:
      'КТМ 125 DUKE - не игрушка. Это настоящий мотоцикл, максимально укомплектован и зачастую лучше оборудован, чем многие мотоциклы более высокого уровня. В 2013 году он был первым 125 кубовым мотоциклом, получившим ABS. И теперь он снова вырвался вперед, получив  светодиодную оптику, TFT-дисплей и усовершенствованный 4-тактный одноцилиндровый двигатель с системой впрыска топлива и 6-ступенчатой коробкой передач. При небольшой цене Duke 125 сочетает в себе превосходную производительность, великолепную управляемость и отличную динамику. Без сомнения, КТМ Дюк 125 лучшее сочетание цены и качества в классе! Цена - НЕ ЯВЛЯЕТСЯ ФИКСИРОВАННОЙ, согласуется в договорном порядке.',
    id: 'ktm-125-duke-2022',
    name: 'KTM 125 Duke 2022',
    photolink:
      'http://ktm.by/images/made/images/moto/ktm-125-duke-2018/black/2802/315743_125-Duke-ALL-OVER-THE-WORLD-2021_30313d26236d139f4f81ffc0ef58aaf2.jpg',
    price: '20340 руб',
    titleCard: 'Разжигающий пламя',
  },

  {
    background:
      'http://ktm.by/images/made/images/bg/2800/358716_KTM-Naked-Model-Range_cebf97f0912754aa122e60151c9c0586.jpeg',
    buy: false,
    category: 'naked',
    counter: 1,
    descriptionMoto:
      'Максимальное удовольствие от движения в сочетании с оптимальным соотношением цена/качество и, на первом месте - управляемость. Легкий, как перо, мощный и оснащенный самыми современными техническими решениями, он гарантирует неизменно волнующие ощущения от поездки, где бы вы ни были - в городских джунглях или извилистой лесной дорожке. 390-й Дюк - бескомпромиссный лидер в своем классе, предлагающий наилучшее оснащение, которому могут позавидовать более крупные соперники, и эталонные управляемость и динамику. Ключевые инновации:  Впервые в классе - многоцветный TFT- дисплей с адаптивной подсветкой. Светодиодная фара, своим дизайном и функциональностью идеально подходящая к концепции всего мотоцикла. Система KTM My Ride для управления смартфоном. Увеличенный до 13,4 л объем топливного бака. Новая выхлопная система с боковым расположением. Управление дросселем Ride-By-Wire. Проскальзывающее сцепление. Ну и конечно же напоминаем, что при покупке мотоцикла, у нас в салоне, вы получите именную карту постоянного клиента, которая даст вам возможность приобретать товары и услуги со скидкой. Цена - НЕ ЯВЛЯЕТСЯ ФИКСИРОВАННОЙ, согласуется в договорном порядке. Конфигуратор дополнительного оборудования: https://configurator.ktm.com/m...',
    id: 'ktm-390-duke-2022',
    name: 'KTM 390 DUKE 2022',
    photolink:
      'http://ktm.by/images/made/images/moto/ktm-390-duke-2020-1/white/2920/315806_390-Duke-INDIA-2020-ALL-OVER-THE-WORLD-2021_210611_102144_30313d26236d139f4f81ffc0ef58aaf2.jpg',
    price: '22464 руб',
    titleCard: 'Незаменимый',
  },
  {
    background:
      'http://ktm.by/images/made/images/bg/2713/361975_MY21-KTM-890-DUKE_Action_cebf97f0912754aa122e60151c9c0586.jpeg',
    buy: false,
    category: 'naked',
    counter: 1,
    descriptionMoto:
      'KTM 8 90 DUKE дает новое определение слову «резкий», добавляя дополнительные грани. KTM 890 DUKE, оснащенный рядным двухцилиндровым двигателем объемом 889 куб.см, размещенным на одном из самых легких и компактных шасси, может похвастаться маневренностью, которую вы ожидаете от 600-кубового двигателя, но с мощностью более крупного двигателя. Он не переписывает книгу правил для среднекубатурного "нэйкеда" , он стирает его. Цена - НЕ ЯВЛЯЕТСЯ ФИКСИРОВАННОЙ, согласуется в договорном порядке. Конфигуратор дополнительного оборудования: https://configurator.ktm.com/m...',
    id: 'ktm-890-duke-2022',
    name: 'KTM 890 Duke 2022',
    photolink:
      'http://ktm.by/images/made/images/moto/ktm-790-duke-2018-1/orange/2715/372812_890-Duke-ORANGE-MY21-90-Right_30313d26236d139f4f81ffc0ef58aaf2.jpg',
    price: '42336 руб',
    titleCard: 'Скальпель дорог',
  },
  {
    background:
      'http://ktm.by/images/made/images/bg/2664/351531_KTM-690-ENDURO-R-2021_cebf97f0912754aa122e60151c9c0586.jpeg',
    buy: false,
    category: 'travel',
    counter: 1,
    descriptionMoto:
      'Выезжайте из дома и почувствуйте, как накапливается адреналин, пройдите заключительные повороты асфальта к любимому месту бездорожья. Прицепа не требуется, просто поверните дроссель KTM 690 ENDURO R, уверенный, когда начинается грязь и трасса становится более экстремальной. Легкое шасси, агрессивный стиль и проверенный двигатель 690 LC4 дополняются новейшей электроникой и подвеской WP XPLOR. Осваивайте более сложную местность на этом легком и мощном короле универсальности.',
    id: 'ktm-690-enduro-r-2022',
    name: 'KTM 690 Enduro R 2022',
    photolink:
      'http://ktm.by/images/made/images/moto/ktm-690-enduro-r-2021/orange/2666/350478_690-ENDURO-R-2021_30313d26236d139f4f81ffc0ef58aaf2.jpg',
    price: '42912 руб',
    titleCard: 'Король универсальности',
  },
];

export const state = {
  moto: {
    moto: [
      {
        background:
          'http://ktm.by/images/made/images/bg/2713/361975_MY21-KTM-890-DUKE_Action_cebf97f0912754aa122e60151c9c0586.jpeg',
        buy: false,
        category: 'naked',
        counter: 1,
        descriptionMoto:
          'KTM 8 90 DUKE дает новое определение слову «резкий», добавляя дополнительные грани. KTM 890 DUKE, оснащенный рядным двухцилиндровым двигателем объемом 889 куб.см, размещенным на одном из самых легких и компактных шасси, может похвастаться маневренностью, которую вы ожидаете от 600-кубового двигателя, но с мощностью более крупного двигателя. Он не переписывает книгу правил для среднекубатурного "нэйкеда" , он стирает его. Цена - НЕ ЯВЛЯЕТСЯ ФИКСИРОВАННОЙ, согласуется в договорном порядке. Конфигуратор дополнительного оборудования: https://configurator.ktm.com/m...',
        id: 'ktm-890-duke-2022',
        name: 'KTM 890 Duke 2022',
        photolink:
          'http://ktm.by/images/made/images/moto/ktm-790-duke-2018-1/orange/2715/372812_890-Duke-ORANGE-MY21-90-Right_30313d26236d139f4f81ffc0ef58aaf2.jpg',
        price: '42336 руб',
        titleCard: 'Скальпель дорог',
      },
      {
        background:
          'http://ktm.by/images/made/images/bg/2664/351531_KTM-690-ENDURO-R-2021_cebf97f0912754aa122e60151c9c0586.jpeg',
        buy: false,
        category: 'travel',
        counter: 1,
        descriptionMoto:
          'Выезжайте из дома и почувствуйте, как накапливается адреналин, пройдите заключительные повороты асфальта к любимому месту бездорожья. Прицепа не требуется, просто поверните дроссель KTM 690 ENDURO R, уверенный, когда начинается грязь и трасса становится более экстремальной. Легкое шасси, агрессивный стиль и проверенный двигатель 690 LC4 дополняются новейшей электроникой и подвеской WP XPLOR. Осваивайте более сложную местность на этом легком и мощном короле универсальности.',
        id: 'ktm-690-enduro-r-2022',
        name: 'KTM 690 Enduro R 2022',
        photolink:
          'http://ktm.by/images/made/images/moto/ktm-690-enduro-r-2021/orange/2666/350478_690-ENDURO-R-2021_30313d26236d139f4f81ffc0ef58aaf2.jpg',
        price: '42912 руб',
        titleCard: 'Король универсальности',
      },
    ],
    page: [
      {
        background:
          'http://ktm.by/images/made/images/bg/2473/1290-SUPER-DUKE-R-2020-cover_2f552fd4e6f76672742481ebb95c4027.jpeg',
        categoryTitle: 'KTM Naked Bikes',
        description: 'Городские мотоциклы-стритфайтеры',
        id: 'naked',
        path: '/naked',
      },
      {
        background:
          'http://ktm.by/images/made/images/bg/2877/342521_KTM-450-SX-F-2021_2f552fd4e6f76672742481ebb95c4027.jpeg',
        categoryTitle: 'КТМ МХ',
        description: 'Кроссовые мотоциклы',
        id: 'mx',
        path: '/mx',
      },
      {
        background:
          'http://ktm.by/images/made/images/bg/2556/177270_KTM-350-EXC-F-MY-2018_2f552fd4e6f76672742481ebb95c4027.jpeg',
        categoryTitle: 'Эндуро мотоциклы',
        description: '',
        id: 'enduro',
        path: '/enduro',
      },
    ],
  },
};

export const initialState = { error: null, loading: false, moto: [], page: [] };
