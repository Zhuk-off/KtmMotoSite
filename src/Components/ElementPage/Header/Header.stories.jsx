import { Header } from './Header';
import '../../../index.scss';

export default {
  title: 'Elements/Header',
  component: Header,
  argTypes: {
    length: {
      type: 'number',
      description: 'Количество мотоциклов выведеных на экран',
      defaultValue: 1,
      options: [1, 2, 3, 4, 5],
      control: {
        type: 'select',
      },
    },
  },
};

const Template = (arg) => <Header {...arg} />;

export const Default = Template.bind({});
Default.args = {
  category: 'Эндуро мотоциклы',
  description: 'Эндуро',
  length: 3,
  background:
    'http://ktm.by/images/made/images/bg/2556/177270_KTM-350-EXC-F-MY-2018_2f552fd4e6f76672742481ebb95c4027.jpeg',
};
