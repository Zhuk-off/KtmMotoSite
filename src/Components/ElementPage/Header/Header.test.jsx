import { screen, render } from '@testing-library/react';
import { Header } from './Header';
import { moto, page } from '../../TestHelper/testHelper';

describe('Header', () => {
  it('renders Header with data', () => {
    render(
      <Header
        category={moto.category}
        description={page.description}
        length={1}
        background={moto.background}
      />
    );
    expect(screen.getByText(/модель/)).toBeInTheDocument;
    expect(screen.getByText(/naked/)).toBeInTheDocument;
    expect(screen.getByText(/1/)).toBeInTheDocument;
    expect(screen.getByText(/Городские мотоциклы-стритфайтеры/))
      .toBeInTheDocument;
  });
  it('Header snapshot', () => {
    const header = render(
      <Header
        category={moto.category}
        description={page.description}
        length={1}
        background={moto.background}
      />
    );
    expect(header).toMatchSnapshot();
  });
});
