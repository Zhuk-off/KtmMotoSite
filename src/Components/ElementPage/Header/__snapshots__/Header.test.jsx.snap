// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pageHeading"
        style="background-image: url(http://ktm.by/images/made/images/bg/2713/361975_MY21-KTM-890-DUKE_Action_cebf97f0912754aa122e60151c9c0586.jpeg);"
      >
        <div
          class="pageHeadingText"
        >
          <h1>
            naked
             
          </h1>
          <h2>
            Городские мотоциклы-стритфайтеры
          </h2>
          <h3>
            1
             
            модель
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pageHeading"
      style="background-image: url(http://ktm.by/images/made/images/bg/2713/361975_MY21-KTM-890-DUKE_Action_cebf97f0912754aa122e60151c9c0586.jpeg);"
    >
      <div
        class="pageHeadingText"
      >
        <h1>
          naked
           
        </h1>
        <h2>
          Городские мотоциклы-стритфайтеры
        </h2>
        <h3>
          1
           
          модель
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
